include/circt/Dialect/RTGTest/IR/RTGTestOps.td

```diff
@@ -33,3 +35,21 @@ def CPUDeclOp : RTGTestOp<"cpu_decl", [
 
   let assemblyFormat = "$id attr-dict";
 }
+
+def RegisterOp : RTGTestOp<"reg", [
+  DeclareOpInterfaceMethods<RegisterOpInterface>,
+  DeclareOpInterfaceMethods<InferTypeOpInterface>,
+]> {
+  let summary = "returns a value representing a register";
+  let description = [{
+    This operation creates a value representing the register given as the 'reg'
+    attribute. A register can be a concrete register or a virtual register.
+    Virtual registers will be assigned a concrete register when running register
+    allocation.
```

Comment:
This probably doesn't make sense for a test dialect, but in practice, would you have virtual registers as a separate op that cannot be CSEd, and actual ISA register as something "pure" or maybe even "constant-like"?

Reply:
Good point! Having concrete registers CSE would be very nice. Having separate ops for concrete and virtual registers complicates register allocation a bit but makes elaboration easier. Let me think about that a bit more.